### PR TITLE
feat(extensions): add support for some chrome.management APIs

### PR DESCRIPTION
**Pull request by sentialx**
#### Description of Change
This PR implements missing `extensions::ManagementAPIDelegate` which is being used by `chrome.management` API and also fixes a crash when an extension is attempting to use the API.
It doesn't implement all of the `chrome.management` functionalities, as they need to emit relevant events in `session` or in `session.extensions`, so that developers could set their own behavior. I will probably do that in another PR.

Could this be backported to 9-x-y?

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).
- [x] This is **NOT A BREAKING CHANGE**. Breaking changes may not be merged to master until 11-x-y is branched.

#### Release Notes

Notes: Added support for some `chrome.management` APIs.

### CODE REVIEW

File: docs/api/extensions.md [hunk 1]
```diff
@@ -1,104 +1,112 @@
-# Chrome Extension Support
```
Comment (zcbenz) [inline docs/api/extensions.md:1]: Line ending has been changed in this file.

**Review by zcbenz — CHANGES_REQUESTED** [review]

File: docs/api/extensions.md (hunk 1 above)
Comment (sentialx) [reply docs/api/extensions.md:1]: @zcbenz Seems like the pre-commit hook changes LF to CRLF automatically

Comment (zcbenz) [reply docs/api/extensions.md:1]: I added a change to make sure the files have correct line endings (https://github.com/electron/electron/pull/25134), can you rebase this branch on master?

Comment (sentialx) [reply docs/api/extensions.md]: Done!

Comment (zcbenz) [reply docs/api/extensions.md]: @sentialx Sorry the CI is failing because of forcing line endings, do you mind rebasing to latest master again?

Comment (sentialx) [reply docs/api/extensions.md]: Sure

**Review by zcbenz — APPROVED** [review]

**Comment by sentialx** [comment]
@zcbenz Could it be backported to 9-x-y?

**Comment by zcbenz** [comment]
@sentialx You have to manually backport the PR to 11/10/9 branches, and it would also require approval from @electron/wg-api to get merged.

**Pull request merged**